go_modules/lib/dependabot/go_modules/file_fetcher.rb

```diff
@@ -16,10 +16,16 @@ def self.required_files_message
 
       def ecosystem_versions
         return nil unless go_mod
-
+        go_version_string = go_mod.content.match(/^go\s(\d+\.\d+)/)&.captures&.first || "unknown"
```

Comment:
No need to coerce as this is already only major.minor, patch isn't supported in `go.mod` files.